source/agora/test/ValidatorGeneral.d

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+
+    Contains tests for general situations where validators participate in
+    nominating a block and reaching consensus.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.ValidatorGeneral;
+
+version (unittest):
+
+import agora.test.Base;
+
+/// Situation: A validator is stopped and wiped clean after the network starts.
+///     And then, two blocks are inserted into the ledger. After the validator
+///     restarts, another new block is in the middle of a consensus.
+/// Expectation: The new block is inserted into the ledger because the validator
+///     has started to validate immediately.
+unittest
+{
+    TestConf conf = { validators : 3, full_nodes : 1 };
+    auto network = makeTestNetwork(conf);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+
+    // The node_1 is the validator
+    auto nodes = network.clients;
+    auto node_1 = nodes[0];
+
+    // Create two blocks from the Genesis block
+    auto txs = genesisSpendable().enumerate()
+        .map!(en => en.value.refund(WK.Keys[en.index].address).sign())
+        .array();
+    txs.each!(tx => node_1.putTransaction(tx));
+    network.expectBlock(Height(1), 8.seconds);
+
+    txs = txs.enumerate()
+        .map!(en => TxBuilder(en.value)
+            .refund(WK.Keys[8 + en.index].address)
```

Comment:
By the way the `+8` here and `+16` below is strange and distracts from the test. Why not just refund to the same keys? Or even to just one public key?

It makes no difference but it makes the tests simpler. I was just wondering why it was done.